contracts/0.8.9/StakingRouter.sol

```diff
@@ -964,34 +964,37 @@ contract StakingRouter is AccessControlEnumerable, BeaconChainDepositor, Version
     ) external payable validStakingModuleId(_stakingModuleId) {
         if (msg.sender != LIDO_POSITION.getStorageAddress()) revert AppAuthLidoFailed();
 
+        StakingModule storage stakingModule = _getStakingModuleById(_stakingModuleId);
         uint256 depositsValue = msg.value;
-        if (depositsValue == 0 || depositsValue != _depositsCount * DEPOSIT_SIZE) {
-            revert InvalidDepositsValue(depositsValue);
-        }
 
-        bytes32 withdrawalCredentials = getWithdrawalCredentials();
-        if (withdrawalCredentials == 0) revert EmptyWithdrawalsCredentials();
+        if (depositsValue > 0) {
+            if (depositsValue != _depositsCount * DEPOSIT_SIZE) {
```

Comment:
can be outside `if` (minorish consideration)

Reply:
in general case `_depositsCount ` can be `!= 0` when `depositsValue = 0`, so it's just works like require

Reply:
Yeah, I see